includes/class-sensei-lesson.php

```diff
@@ -363,6 +378,66 @@ public function lesson_prerequisite_meta_box_content() {
 		);
 	}
 
+	/**
+	 * Returns all the lesson prerequisite posts, ordered by modules.
+	 *
+	 * @since 3.14.0
+	 *
+	 * @param int $lesson_id The lesson id.
+	 * @param int $course_id The course id.
+	 *
+	 * @return WP_Post[]
+	 */
+	private function get_prerequisites( int $lesson_id, int $course_id ): array {
+		$lesson_status = [ 'publish', 'draft', 'future' ];
+		$modules       = Sensei()->modules->get_course_modules( $course_id );
+		$lessons       = Sensei()->modules->get_none_module_lessons( $course_id, $lesson_status );
+
+		// If the course has modules, make sure the lessons order is correct.
+		if ( $modules ) {
+			$in_module_lessons = [];
+			foreach ( $modules as $module_term ) {
+				$module_lessons_query = Sensei()->modules->get_lessons_query(
+					$course_id,
+					$module_term->term_id,
+					$lesson_status
+				);
+
+				$in_module_lessons = array_merge( $in_module_lessons, $module_lessons_query->get_posts() );
+			}
+
+			$lessons = array_merge( $in_module_lessons, $lessons );
+		}
+
+		// Exclude the lesson that we are getting the prerequisites for.
+		$lessons = array_filter(
+			$lessons,
+			function( $lesson ) use ( $lesson_id ) {
+				return $lesson->ID !== $lesson_id;
+			}
+		);
+
+		return $lessons;
+	}
+
+	/**
+	 * Handles the prerequisite meta box ajax request by outputting the box content HTML.
+	 *
+	 * @since 3.14.0
+	 */
+	public function handle_get_prerequisite_meta_box_content() {
+		check_ajax_referer( 'get_prerequisite_meta_box_content_nonce', 'security' );
+
+		if ( isset( $_GET['lesson_id'] ) && isset( $_GET['course_id'] ) ) {
```

Comment:
What's your opinion on adding a `current_user_can` capability check here?

Reply:
Makes sense. I guess the nonce is not that reliable because we load the localized script on different pages. Should we use the `edit_lessons` capability?

Reply:
I was on the fence about whether we need one because we're only reading data here, right? I think I've only ever seen read capability checks done here if it's `read_private_xxx`, but not necessarily for `read_xxx`. I was actually going to let you decide. 😄  But if we do add it I guess we would use `read_lesson`.

Reply:
Yeah, it's a bit overkill, and I'm not a big fan of futureproofing (can lead to more problems than it solves). For this particular case, I don't see a big reason to add authorization, the token check should be enough, but I guess we need to decide if we should make this a practice for all other handlers. :)

Reply:
> I don't see a big reason to add authorization

Sounds good to me!

> For example, we can say - all handlers should have an authorization check.

We'd be breaking our own rule in this case then, wouldn't we? 😄 There will always be exceptions though. But yeah, as long as we call them "guidelines" and not "hard-and-fast rules", that sounds reasonable.

Reply:
Got it, skipping the authorization for this one. 🙂